### PR TITLE
feat: Add Data catalog components

### PR DESCRIPTION
**Pull request by tumido**
Enables Data catalog components in Dashboard

Requires:
https://github.com/opendatahub-io/odh-manifests/pull/228
https://github.com/opendatahub-io/odh-manifests/pull/227

### CODE REVIEW

**Comment by tumido** [comment]
@cfchase this can be reviewed and merged now, we have datacatalog fully merged to the `odh-manifests` now

**Comment by tumido** [comment]
cc @nakfour @vpavlin @LaVLaS

File: routes/api/components/available-components.js [hunk 1]
```diff
@@ -84,4 +84,22 @@ module.exports = [
...
+    img: "images/hue.svg",
+    docsLink: "https://docs.gethue.com/",
+  },
+  {
+    key: "thriftserver",
+    label: "Spark SQL Thrift Server",
+    description: "Expose Spark data frames modeled as Hive tables through a JDBC connection",
+    kfdefApplications: ["thriftserver"],
+    route: "thriftserver",
+    img: "images/spark.svg",
```
Comment (cfchase) [inline routes/api/components/available-components.js:102]: @tumido This is probably ok for now, but I think we don't really want 2 tiles with the exact same icon.  Perhaps we can engage @kywalker-rh about what icon he'd like to use.  Maybe we can overlay a SQL icon over Spark or something.

**Review by cfchase — APPROVED** [review]

File: routes/api/components/available-components.js (hunk 1 above)
Comment (tumido) [reply routes/api/components/available-components.js:102]: Yeah.. the problem is that Thrift Server by itself doesn't have it's own icon/logo. They use the Spark logo instead. Which is unfortunate. :slightly_smiling_face:

Comment (kywalker-rh) [reply routes/api/components/available-components.js:102]: Thanks for adding me @cfchase. You're right, its not ideal to have more than one tile with the same icon, but its probably okay for now. Thinking it through, if we had more than one product from the same company, say two services from Red Hat, or Anaconda, they would probably have the same icon too, though, so maybe its not a big deal at all. But I will ask around and see if there are other instances on other projects that we'd want to emulate for a situation like this.

Comment (tumido) [reply routes/api/components/available-components.js:102]: FFR, tracked at https://github.com/opendatahub-io/odh-dashboard/issues/36

**Pull request merged**